kong_test.go

```diff
@@ -1542,3 +1542,106 @@ func TestPassthroughCmdOnlyStringArgs(t *testing.T) {
 	_, err := kong.New(&cli)
 	require.EqualError(t, err, "<anonymous struct>.Command: passthrough command command [<args> ...] must contain exactly one positional argument of []string type")
 }
+
+func TestStringPointer(t *testing.T) {
+	var cli struct {
+		Foo *string
+	}
+	k, err := kong.New(&cli)
+	require.NoError(t, err)
+	require.NotNil(t, k)
+	ctx, err := k.Parse([]string{"--foo", "wtf"})
+	require.NoError(t, err)
+	require.NotNil(t, ctx)
+	require.NotNil(t, cli.Foo)
+	require.Equal(t, "wtf", *cli.Foo)
+}
+
+func TestStringPointerNoValue(t *testing.T) {
+	var cli struct {
+		Foo *string
+	}
+	k, err := kong.New(&cli)
+	require.NoError(t, err)
+	require.NotNil(t, k)
+	ctx, err := k.Parse([]string{})
+	require.NoError(t, err)
+	require.NotNil(t, ctx)
+	require.Nil(t, cli.Foo)
+}
+
+func TestStringPointerDefault(t *testing.T) {
+	var cli struct {
+		Foo *string `default:"stuff"`
+	}
+	k, err := kong.New(&cli)
+	require.NoError(t, err)
+	require.NotNil(t, k)
+	ctx, err := k.Parse([]string{})
+	require.NoError(t, err)
+	require.NotNil(t, ctx)
+	require.NotNil(t, cli.Foo)
+	require.Equal(t, "stuff", *cli.Foo)
+}
+
+func TestStringPointerAliasNoValue(t *testing.T) {
+	type Foo string
+	var cli struct {
+		F *Foo
+	}
+	k, err := kong.New(&cli)
+	require.NoError(t, err)
+	require.NotNil(t, k)
+	ctx, err := k.Parse([]string{})
+	require.NoError(t, err)
+	require.NotNil(t, ctx)
+	require.Nil(t, cli.F)
+}
+
+func TestStringPointerAlias(t *testing.T) {
+	type Foo string
+	var cli struct {
+		F *Foo
+	}
+	k, err := kong.New(&cli)
+	require.NoError(t, err)
+	require.NotNil(t, k)
+	ctx, err := k.Parse([]string{"--f=value"})
+	require.NoError(t, err)
+	require.NotNil(t, ctx)
+	require.NotNil(t, cli.F)
+	require.Equal(t, Foo("value"), *cli.F)
+}
+
+func TestStringPointerEmptyValue(t *testing.T) {
+	var cli struct {
+		F *string
+		G *string
+	}
+	k, err := kong.New(&cli)
+	require.NoError(t, err)
+	require.NotNil(t, k)
+	ctx, err := k.Parse([]string{"--f", "", "--g="})
+	require.NoError(t, err)
+	require.NotNil(t, ctx)
+	require.NotNil(t, cli.F)
+	require.NotNil(t, cli.G)
+	require.Equal(t, "", *cli.F)
+	require.Equal(t, "", *cli.G)
+}
+
+func TestIntPtr(t *testing.T) {
```

Comment:
Nice, I like the change. Can you convert these tests to table driven tests though, please?

Reply:
Actually don't worry about converting the tests, it's fine as-is.